src/main/java/org/apache/maven/plugins/pmd/PmdReport.java

```diff
@@ -561,7 +563,7 @@ private String determineAuxClasspath() throws MavenReportException
                 }
                 TransformableFilter filter = new AndFilter( Arrays.asList(
                         new ExclusionsFilter( exclusionPatterns ),
-                        includeTests ? ScopeFilter.including( "test" ) : ScopeFilter.including( "compile" )
+                        includeTests ? ScopeFilter.including( "compile", "test" ) : ScopeFilter.including( "compile" )
```

Comment:
This looks like this is the actual fix. We probably should also consider here the scope "provided". The goal is, that we get the classpath that is used for compilation...

Reply:
The missing provided scope might explain https://issues.apache.org/jira/browse/MPMD-317